### PR TITLE
adjust touch logic for non-interactive images

### PR DESCRIPTION
**Pull request by abrin**
CanvasPanel can use a static-preset, but set interactive to `false`.  This works _ok_ in many cases, but falls down in a few.  This allows us to adjust the interactivity from the css level by moving the pointer-events logic into it’s own css class that’s  assigned based on the interactive prop

### CODE REVIEW

File: src/modules/react-reconciler/Atlas.tsx [hunk 1]
```diff
@@ -445,6 +445,11 @@ export const Atlas: React.FC<
 
   const { height: _, width: __, ...canvasProps } = restProps;
   const widthClassName = useClassname([restProps.width, restProps.height]);
+  let isInteractive = true;
+  // if we have a render preset and that render preset sets interactive to false, then... disable it
+  if (renderPreset && Array.isArray(renderPreset) && renderPreset.length > 1 && (renderPreset[1] as any).interactive === false) {
+    isInteractive = false;
+  }
```
Comment (abrin) [inline src/modules/react-reconciler/Atlas.tsx:452]: is this a logical assumption, by default it's interactive, but if someone explicitly sets the presets to disable the interactivity, then it isn't

File: src/modules/react-reconciler/Atlas.tsx [hunk 2]
```diff
@@ -483,7 +488,11 @@ export const Atlas: React.FC<
           ref={refs.canvas as any}
         />
       )}
-      <Container className="atlas-overlay" style={{ ...(overlayStyle || {}) }} ref={refs.overlay as any}>
+      
+      <Container className={['atlas-overlay', isInteractive ? 'atlas-overlay--interactive' : null]
```
Comment (bulbil) [inline src/modules/react-reconciler/Atlas.tsx:492]: Elsewhere Stephen uses `''` instead of `null` -- I wonder if we should be consistent

**Comment by bulbil** [comment]
This makes a lot of sense to me, being a bit more selective of where `pointer-events: none` gets applied. 

I noticed that `atlas-static-image` also has `pointer-events: none`-- I was thinking that ought to be removed as well

File: src/modules/overlay-renderer/overlay-renderer.ts [hunk 3]
```diff
@@ -66,8 +66,8 @@ export class OverlayRenderer implements Renderer {
         position: absolute;
         transform-origin: 0px 0px;
       `),
-      interactive: this.stylesheet.addStylesheet(`pointer-events: all`),
-      nonInteractive: this.stylesheet.addStylesheet(`pointer-events: none`),
+      interactive: this.stylesheet.addStylesheet(`pointer-events: none`),
+      nonInteractive: this.stylesheet.addStylesheet(`pointer-events: all`),
```
Comment (abrin) [inline src/modules/overlay-renderer/overlay-renderer.ts:70]: @stephenwf I noticed these should be changed, but wasn't sure if there was a good way to test this and the 'atlas-static-image' class below too

File: src/modules/react-reconciler/Atlas.tsx [hunk 4]
```diff
@@ -483,7 +487,11 @@ export const Atlas: React.FC<
...
```
Comment (abrin) [inline src/modules/react-reconciler/Atlas.tsx:491]:
```suggestion
      <Container className={['atlas-overlay', isInteractive ? 'atlas-overlay--interactive' : '']
```

File: src/modules/react-reconciler/Atlas.tsx [hunk 5]
```diff
@@ -526,8 +534,10 @@ export const Atlas: React.FC<
         .atlas-canvas:focus-visible, .atlas-canvas-container:focus-visible { outline: var(--atlas-focus, 2px solid darkorange) }
         .atlas-static-preset { touch-action: inherit; }
         .atlas-static-container { position: relative; overflow: hidden; flex: 1 1 0px; }
-        .atlas-overlay { position: absolute; top: 0; left: 0; pointer-events: none; overflow: hidden; }
-        .atlas-static-image { position: absolute; pointer-events: none; user-select: none; transform-origin: 0px 0px; }
+        .atlas-overlay { position: absolute; top: 0; left: 0; none; overflow: hidden; }
+        /** setting the pointer events to none means that Atlas will own the touch and mousewheel events **/
+        .atlas-overlay--interactive { pointer-events: none; }
+        .atlas-static-image { position: absolute; user-select: none; transform-origin: 0px 0px; }
```
Comment (abrin) [inline src/modules/react-reconciler/Atlas.tsx:540]: I wasn't sure where best to test this, but removing it didn't seem to change any of the stories

File: src/modules/overlay-renderer/overlay-renderer.ts (hunk 3 above)
Comment (stephenwf) [reply src/modules/overlay-renderer/overlay-renderer.ts:70]: Hmm, are you sure? The interactive property was intended to indicate if the HTML in the overlay is interactive (e.g. if there was text you could select it, or a button you could click it). If it's marked as `nonInteractive` then it the _atlas_ events such as `onClick` would still work.

**Comment by stephenwf** [comment]
One thing to test might be the box-resizing on the homepage annotations story. I think that uses `interactive` at the overlay level. 

Changes look good, I think I've confused things by using the word `interactive` to mean two different things in two different contexts! (Is the canvas interactive, is the HTML painted on the canvas interactive). I will try to get this pulled down later this week and test with some of the annotation stuff.

File: src/modules/overlay-renderer/overlay-renderer.ts (hunk 3 above)
Comment (abrin) [reply src/modules/overlay-renderer/overlay-renderer.ts:70]: no, not sure at all, hence the question.  Will swap it back

**Comment by abrin** [comment]
> One thing to test might be the box-resizing on the homepage annotations story. I think that uses `interactive` at the overlay level.
> 
> Changes look good, I think I've confused things by using the word `interactive` to mean two different things in two different contexts! (Is the canvas interactive, is the HTML painted on the canvas interactive). I will try to get this pulled down later this week and test with some of the annotation stuff.

@stephenwf So... the homepage renderer story gets at the core of the question, good point -- should we assume that if the `static-renderer` preset is set **at all** that we should not enable any interactions?  Or, does the `interactive` property that can be set on the renderer govern this?

File: src/modules/overlay-renderer/overlay-renderer.ts (hunk 3 above)
Comment (abrin) [reply src/modules/overlay-renderer/overlay-renderer.ts]: reverted

**Comment by stephenwf** [comment]
I think the changes to the static preset (from the storybook) are fine. I don't know of any uses of the static preset + annotations. Are you happy for me to merge and release?

**Comment by abrin** [comment]
Hi Stephen, yes please & thank you!

**Comment by stephenwf** [comment]
`v2.2.8` should be published in a few minutes 👍

**Pull request merged**